Core.ApplicationServices/Generic/Write/AssignmentUpdateService.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Core.Abstractions.Extensions;
+using Core.Abstractions.Types;
+using Core.ApplicationServices.OptionTypes;
+using Core.DomainModel;
+using Core.DomainServices.Generic;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Core.ApplicationServices.Generic.Write
+{
+    public class AssignmentUpdateService : IAssignmentUpdateService
+    {
+        private readonly IOptionResolver _optionResolver;
+        private readonly IEntityIdentityResolver _entityIdentityResolver;
+
+        public AssignmentUpdateService(IOptionResolver optionResolver, IEntityIdentityResolver entityIdentityResolver)
+        {
+            _optionResolver = optionResolver;
+            _entityIdentityResolver = entityIdentityResolver;
+        }
+
+        public Maybe<OperationError> UpdateIndependentOptionTypeAssignment<TDestination, TOption>(
+            TDestination destination,
+            Guid? optionTypeUuid,
+            Action<TDestination> onReset,
+            Func<TDestination, TOption> getCurrentValue,
+            Action<TDestination, TOption> updateValue)
+                where TOption : OptionEntity<TDestination>
+                where TDestination : IOwnedByOrganization
+        {
+            if (optionTypeUuid == null)
+            {
+                onReset(destination);
+            }
+            else
+            {
+                var optionType = _optionResolver.GetOptionType<TDestination, TOption>(destination.Organization.Uuid, optionTypeUuid.Value);
+                if (optionType.Failed)
+                {
+                    return new OperationError($"Failure while resolving {typeof(TOption).Name} option:{optionType.Error.Message.GetValueOrEmptyString()}", optionType.Error.FailureType);
+                }
+
+                var option = optionType.Value;
+                var currentValue = getCurrentValue(destination);
+                if (option.available == false && (currentValue == null || currentValue.Uuid != optionTypeUuid.Value))
+                {
+                    return new OperationError($"The changed {typeof(TOption).Name} points to an option which is not available in the organization", OperationFailure.BadInput);
+                }
+
+                updateValue(destination, option.option);
+            }
+
+            return Maybe<OperationError>.None;
+        }
+
+        public Maybe<OperationError> UpdateMultiAssignment<TDestination, TAssignment, TKey>(
```

Comment:
UpdateUniqueMultiAssignment tror jeg vi skal kalde den for det er jo faktisk en forudsætning for at den virker og det tjekker den også på inden den laver delta (det er for at delta kan lade sig gøre)